tests/testthat/test-submodule.R

```diff
@@ -0,0 +1,113 @@
+context("submodule.R")
+
+test_that("parse_submodules works with a single submodule", {
+  x <-
+'[submodule "foobar"]
+  path = baz
+  url = http://foo/bar'
+
+  expect_equal(
+    parse_submodules(x),
+    data.frame(
+      submodule = "foobar",
+      path = "baz",
+      url = "http://foo/bar",
+      branch = NA_character_,
+      stringsAsFactors = FALSE))
+})
+
+test_that("parse_submodules works multiple submodules", {
+  y <-
+'[submodule "foobar"]
+  path = baz
+  url = http://foo/bar
+
+[submodule "foofoo"]
+  path = bunny
+  url = http://little/bunny/foofoo
+  branch = forest'
+
+  expect_equal(
+    parse_submodules(y),
+    data.frame(
+      submodule = c("foobar", "foofoo"),
+      path = c("baz", "bunny"),
+      url = c("http://foo/bar", "http://little/bunny/foofoo"),
+      branch = c(NA_character_, "forest"),
+      stringsAsFactors = FALSE))
+})
+
+test_that("parse_submodules warns and returns empty for invalid submodules", {
+  x <-
+'[submodule "foobar"]
+  path = baz'
+
+  expect_warning(regexp = "Invalid submodule definition",
+    expect_equal(
+      parse_submodules(x),
+      list()
+    )
+  )
+
+  y <-
+'[submodule "foobar"]
+  path = baz
+
+[submodule "foofoo"]
+  path = bunny
+  url = http://little/bunny/foofoo'
+
+  expect_warning(regexp = "Invalid submodule definition",
+    expect_equal(
+      parse_submodules(y),
+      list()
+    )
+  )
+
+  z <- '
+  # [submodule "foobar"] this one is commented out
+  # path = baz
+  # url = https://foo/bar'
+
+  expect_equal(
+    parse_submodules(z),
+    list()
+  )
+})
+
+test_that("Can install a repo with a submodule", {
+  dir <- tempfile()
```

Comment:
Need to skip if no system git.  (If we are skipping the other tests that rely on system git.)